src/mixins/filing-template-mixin.ts

```diff
@@ -925,35 +939,36 @@ export default class FilingTemplateMixin extends DateMixin {
     } else {
       // store default data
       this.setNameRequest({
-        legalType: entitySnapshot.businessInfo.legalType,
-        nrNum: entitySnapshot.businessInfo.nrNumber
+        legalType: this.getOriginalLegalType,
+        nrNum: this.getOriginalNrNumber
       } as any)
-      this.setNameRequestLegalName(entitySnapshot.businessInfo.legalName)
+      this.setNameRequestLegalName(this.getOriginalLegalName)
     }
 
     // store Name Translations
     this.setNameTranslations(cloneDeep(
       this.mapNameTranslations(filing.alteration.nameTranslations) ||
-      this.mapNameTranslations(entitySnapshot.nameTranslations) ||
+      this.mapNameTranslations(entitySnapshot?.nameTranslations) ||
       []
     ))
 
     // store Provisions Removed
     if (filing.alteration.provisionsRemoved) this.setProvisionsRemoved(true)
 
     // store Office Addresses **from snapshot** (because we don't change office addresses in an alteration)
-    this.setOfficeAddresses(cloneDeep(entitySnapshot.addresses))
+    this.setOfficeAddresses(cloneDeep(entitySnapshot?.addresses || null))
```

Comment:
In the future, I think it would be better to check if addresses exists before setting them (and same for all other data below). That way, we can leave the initial values in the store and don't need to re-assign those values here as fallbacks. (Initial values in 1 place only.)

Reply:
Got it, noted! Let me know if you want me to change that here.

Reply:
Not here. Future.